transaction.go

```diff
@@ -165,7 +165,13 @@ func TransactionFromBytes(data []byte) (interface{}, error) { // nolint
 		}
 	}
 
-	tx.transactionIDs.locked = true
+	if tx.transactionIDs._Length() > 0 {
```

Comment:
Maybe we should create a method `LockableList._LockIfNotEmpty()` so you don't have to repeat yourself?